### PR TITLE
Extract :core-api project

### PR DESCRIPTION
**Pull request by lptr**
Extract a relatively small project (<400 files, 7 KLOC) from `:core` with most of the Gradle public API that is not already exposed as part of `:base-services` and `:model-core`.

Other projects that require working with tasks or file collections etc. can now depend on `:core-api` instead of the whole of `:core`.

One major portion of the public API is left in `:core` for now: task implementations. These types depend on `AbstractTask`, that in turn depends on a lot of other stuff. We could extract these classes later into `:core-tasks` or some similar project.

### CODE REVIEW

**Comment by lptr** [comment]
Started branch build: https://builds.gradle.org/viewQueued.html?itemId=7638591

**Comment by lptr** [comment]
Branch accept build: https://builds.gradle.org/viewQueued.html?itemId=7647656

File: subprojects/core-api/core-api.gradle [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    implementation libraries.ant
+    implementation libraries.commons_io
+    implementation libraries.commons_lang
+    implementation libraries.jcip
+
+    testFixturesImplementation project(":internalTesting")
+}
+
+useTestFixtures(project: ":logging")
+//useClassycle()
```
Comment (oehme) [inline subprojects/core-api/core-api.gradle]: ⭕️ Remove commented out code

Comment (lptr) [inline subprojects/core-api/core-api.gradle]: This is what we do elsewhere, too, to signal that this should eventually be enabled.

**Comment by lptr** [comment]
I don't think it's a "model": it's all the interfaces to program against (sans `DefaultTask`). I think calling it an API is the right thing to do. We could find something better to replace the `:core` part, though. However, I have no good candidate right now. Let's rename it if we find something better later.

**Pull request merged**